nautobot/extras/models/jobs.py

```diff
@@ -1321,21 +1326,32 @@ def create_schedule(
         Parameters:
             job_model (JobModel): The job model instance.
             user (User): The user who is scheduling the job.
-            name (str, optional): The name of the scheduled job. Defaults to None.
-            start_time (datetime, optional): The start time for the job. Defaults to None.
-            interval (JobExecutionType, optional): The interval type for the job execution.
+            name (str): The name of the scheduled job. Automatically derived from the job_model and start_time if unset.
+            start_time (datetime): The start time for the job. Defaults to the current time if unset.
+            interval (JobExecutionType): The interval type for the job execution.
                 Defaults to JobExecutionType.TYPE_IMMEDIATELY.
-            crontab (str, optional): The crontab string for the schedule. Defaults to "".
-            profile (bool, optional): Flag indicating whether to profile the job. Defaults to False.
-            approval_required (bool, optional): Flag indicating if approval is required. Defaults to False.
-            task_queue (str, optional): The task queue for the job. Defaults to None, which will use the configured default celery queue.
-            ignore_singleton_lock (bool, optional): Flag indicating whether to ignore singleton locks. Defaults to False.
+            crontab (str): The crontab string for the schedule. Defaults to "".
+            profile (bool): Flag indicating whether to profile the job. Defaults to False.
+            approval_required (bool): Flag indicating if approval is required. Defaults to False.
+            job_queue (JobQueue): The Job queue to use. If unset, use the configured default celery queue.
+            task_queue (str): The queue name to use. **Deprecated, prefer `job_queue`.**
+            ignore_singleton_lock (bool): Whether to ignore singleton locks. Defaults to False.
             **job_kwargs: Additional keyword arguments to pass to the job.
 
         Returns:
             ScheduledJob instance
         """
 
+        if job_queue is not None and task_queue is not None and job_queue.name != task_queue:
+            raise ValueError("task_queue and job_queue are mutually exclusive")
+        if job_queue is not None and task_queue is None:
+            task_queue = job_queue.name
+        elif task_queue is not None and job_queue is None:
+            job_queue = JobQueue.objects.get(name=task_queue)
```

Comment:
Might be gold-plating here but should we do a `try/except` here to catch in case the job queue `DoesNotExist`?

Reply:
I was thinking it'd be better to fail outright rather than have surprising behavior in the case of user error.